src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -915,6 +915,17 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
             RefKind refArg = arguments.RefKind(arg);
             RefKind refParm = parameter.RefKind;
 
+            if (arguments.IsExtensionMethodThisArgument(arg))
+            {
+                Debug.Assert(refArg == RefKind.None);
+                if (refParm == RefKind.Ref || refParm == RefKind.RefReadOnly)
```

Comment:
>refParm == RefKind.RefReadOnly [](start = 46, length = 30)

It looks like we are not including this condition in OverloadResolution.IsApplicable